### PR TITLE
Improve performance by caching data and avoiding a memory leak

### LINKED ISSUE #123: Mobile page reload bug
When navigating to the explore page on mobile, the page reloads and can throw a browser reload error sometimes (often on Safari). Not totally sure what is causing this - It happens when navigating from the crop overview page, or just loading the page straight from a URL.

We have some components which try to access data, sometimes before its loaded - we might try handling these more smoothly at a high level (for example don't try to render anything on the sidebar until data is ready) and see if that helps. Its probably worth doing that anyway.

### PR DESCRIPTION
**Pull request by ebrelsford**
I believe this closes #123 or greatly reduces the issue.

In my investigation I noticed a few things:
 * we re-join data on each map load which is slow,
 * we were not `remove`ing mapbox gl instances when leaving the map page, and
 * the SVGs used for the crop cards are a bit heavy (nearly 250k DOM nodes in total)
 * we could further improve performance by loading the map once in the background and overlaying pages on top of it rather than continuously removing + reloading the map if you're going back and forth in the app

I tried to address the first two here. In fully removing the SVGs locally (the 3rd point) and testing performance was improved more. If it isn't too hard to convert these to PNGs I'd be curious to see the result but I don't think it's a dealbreaker.

I'll highlight where the first two points are addressed inline below.

### CODE REVIEW

File: vacs-map-app/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "d3": "^7.8.5",
     "d3-geo-projection": "^4.0.0",
     "jsdom": "^22.1.0",
-    "mapbox-gl": "^3.0.0-rc.2",
+    "mapbox-gl": "^3.0.1",
```
Comment (ebrelsford) [inline vacs-map-app/package.json:23]: I noticed the Mapbox GL JS has a stable release, figured it was worth using.

File: vacs-map-app/src/components/BaseMap.vue [hunk 2]
```diff
@@ -47,6 +47,10 @@ onMounted(() => {
   if (mapPadding.value) map.value.setPadding(mapPadding.value)
 })
 
+onUnmounted(() => {
+  map.value.remove();
+});
```
Comment (ebrelsford) [inline vacs-map-app/src/components/BaseMap.vue:52]: This was a memory leak--Mapbox GL was holding onto resources even though we've navigated away from the map.

File: vacs-map-app/src/components/GridSource.vue [hunk 3]
```diff
@@ -28,40 +28,15 @@ const { id, map, mapReady } = toRefs(props)
...
-  }
-  return gridData.value
-})
-
-const joinedGeoJson = computed(() => {
-  return featureCollection(
-    joinedData.value.map((row) => point([row.X, row.Y], row, { id: row.id }))
-  )
-})
-
```
Comment (ebrelsford) [inline vacs-map-app/src/components/GridSource.vue:64]: As my TODO suggested, this has been moved.

File: vacs-map-app/src/stores/joinedCropData.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        {
+          ...g,
+          ...yieldMap[id],
+        },
+        { id }
+      );
+    });
+
+    return Object.freeze(featureCollection(points))
+  });
```
Comment (ebrelsford) [inline vacs-map-app/src/stores/joinedCropData.js:35]: This is an optimized join that now just happens once app-wide. It previously took ~1s and now takes ~250ms locally.

**Review by mwbernard — APPROVED** [review]
Thanks for this, (3) should now hopefully be taken care of in #125 as well!

**Pull request merged**